src/server/client.ts

```diff
@@ -528,7 +528,32 @@ namespace ts.server {
         }
 
         getDocumentHighlights(fileName: string, position: number): DocumentHighlights[] {
-            throw new Error("Not Implemented Yet.");
+            let { line, offset } = this.positionToOneBasedLineOffset(fileName, position);
+            let args: protocol.FileLocationRequestArgs = { file: fileName, line, offset };
+
+            let request = this.processRequest<protocol.DocumentHighlightsRequest>(CommandNames.DocumentHighlights, args);
+            let response = this.processResponse<protocol.DocumentHighlightsResponse>(request);
+
+            let _self = this;

```

Comment:
If you have to capture `this` it might not be worth it. I'd convert it back to an arrow function.

Also, why `_self` instead of `self`?